framework/src/mesh/PatternedMesh.C

```diff
@@ -112,51 +101,54 @@ PatternedMesh::clone() const
 void
 PatternedMesh::buildMesh()
 {
-  // stitch_meshes() is only implemented for ReplicatedMesh.  So make sure
-  // we have one here before continuing.
-  ReplicatedMesh * the_mesh = dynamic_cast<ReplicatedMesh *>(&getMesh());
-
-  if (!the_mesh)
-    mooseError("Error, PatternedMesh calls stitch_meshes() which only works on ReplicatedMesh.");
-  else
-  {
-    BoundaryID left = getBoundaryID(getParam<BoundaryName>("left_boundary"));
-    BoundaryID right = getBoundaryID(getParam<BoundaryName>("right_boundary"));
-    BoundaryID top = getBoundaryID(getParam<BoundaryName>("top_boundary"));
-    BoundaryID bottom = getBoundaryID(getParam<BoundaryName>("bottom_boundary"));
-
-    // Build each row mesh
-    for (unsigned int i = 0; i < _pattern.size(); i++)
-      for (unsigned int j = 0; j < _pattern[i].size(); j++)
+  // Local pointers to simplify algorithm
+  std::vector<ReplicatedMesh *> row_meshes;
+  row_meshes.reserve(_pattern.size());
+  // First row is the original mesh
+  row_meshes.push_back(_original_mesh);
+  // Copy the remaining pointers into the local vector
+  std::transform(_row_meshes.begin(), _row_meshes.end(), std::back_inserter(row_meshes),
```

Comment:
Really?

This looks like Javascript-esque obfuscation.  Why not just do a loop?

There is nothing wrong with a plain old loop to get things done!